### PR TITLE
[ID-343] Refactor id7:reflow to (partially) be replaced with MediaQueryList listener 

### PR DESCRIPTION
**Pull request by lol768**
WIP because IE11.

### CODE REVIEW

**Comment by lol768** [comment]
This appears to work as-is, in IE11 on Windows 10:

image


I'm unsure why it wasn't a PITA.

File: js/screen-sizes.js [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
-    name: 'md', test() { return FeatureDetect.mq('only all and (min-width: 992px)'); }, container: 970,
+    name: 'md', matcher() { return window.matchMedia('(min-width: 992px)'); }, container: 970,
   },
   {
-    name: 'sm', test() { return FeatureDetect.mq('only all and (min-width: 768px)'); }, container: 750,
+    name: 'sm', matcher() { return window.matchMedia('(min-width: 768px)'); }, container: 750,
   },
   {
-    name: 'xs', test() { return true; },
+    name: 'xs', matcher() { return window.matchMedia('(max-width: 768px)'); },
```
Comment (matmannion) [inline js/screen-sizes.js]: off-by-one here - should be max 767

File: js/reflow-event.jquery.js [hunk 2]
```diff
@@ -17,24 +17,40 @@ class ReflowEvent {
...
+
+    this.onScreenResize();
   }
 
   wireEventHandlers() {
-    $(window).on('resize.id7.reflow.onScreenResize', $.proxy(this.onScreenResize, this));
+    // Get all possible screen sizes from screen-sizes.js
+    // register an event listener on every MediaQueryList
+    const allSizes = allScreenSizes();
+    for (let i = 0; i < allSizes.length; i += 1) {
```
Comment (matmannion) [inline js/reflow-event.jquery.js]: allSizes.forEach((config) => ...

File: js/screen-sizes.js [hunk 3]
```diff
@@ -1,22 +1,25 @@
...
-import FeatureDetect from './feature-detect';
 
 const ScreenSizes = [
   {
-    name: 'lg', test() { return FeatureDetect.mq('only all and (min-width: 1200px)'); }, container: 1170,
+    name: 'lg', matcher() { return window.matchMedia('(min-width: 1200px)'); }, container: 1170,
   },
   {
-    name: 'md', test() { return FeatureDetect.mq('only all and (min-width: 992px)'); }, container: 970,
+    name: 'md', matcher() { return window.matchMedia('(min-width: 992px)'); }, container: 970,
```
Comment (matmannion) [inline js/screen-sizes.js]:
```suggestion
    name: 'md', matcher() { return window.matchMedia('(min-width: 992px) and (max-width: 1199px)'); }, container: 970,
```

File: js/screen-sizes.js [hunk 4]
```diff
@@ -1,22 +1,25 @@
...
   },
   {
-    name: 'sm', test() { return FeatureDetect.mq('only all and (min-width: 768px)'); }, container: 750,
+    name: 'sm', matcher() { return window.matchMedia('(min-width: 768px)'); }, container: 750,
```
Comment (matmannion) [inline js/screen-sizes.js]:
```suggestion
    name: 'sm', matcher() { return window.matchMedia('(min-width: 768px) and (max-width: 991px)'); }, container: 750,
```

**Comment by lol768** [comment]
Seems to do the trick now that the media queries are mutually exclusive (the ones that matter, anyway).

Just double checking in IE11 again, then I'll merge.

**Pull request merged**